spec/json_spec/matchers/have_json_type_spec.rb

```diff
@@ -56,8 +56,14 @@
   it "provides a failure message for negation" do
     matcher = have_json_type(Numeric)
     matcher.matches?(%(10))
-    matcher.failure_message_when_negated.should   eq "Expected JSON value type to not be Numeric, got Fixnum"
-    matcher.failure_message_for_should_not.should eq "Expected JSON value type to not be Numeric, got Fixnum" # RSpec 2 interface
+    klass = if 1.class == Integer
+      'Integer'
+    else
+      'Fixnum'
+    end
+
+    matcher.failure_message_when_negated.should   eq "Expected JSON value type to not be Numeric, got #{klass}"
+    matcher.failure_message_for_should_not.should eq "Expected JSON value type to not be Numeric, got #{klass}" # RSpec 2 interface
```

Comment:
Rather than the fork, maybe we can change line 58 to use a float instead and just tweak the assertion. The intent of this test would still be asserted. Thoughts?